triemux/mux.go

```diff
@@ -107,6 +123,21 @@ func (mux *Mux) RouteChecksum() []byte {
 	return mux.checksum.Sum(nil)
 }
 
+func (e muxEntry) methodSupported(method string) bool {
+	if len(e.methods) == 0 {
+		// If routes don't specify the methods they allow, we assume they
+		// allow all methods.
+		return true
+	}
+	return contains(e.methods, method)
```

Comment:
I think we'd also want a condition in here that returns true if we're given a HEAD and GET is a method. I don't think we'd want every GET to be registered with a corresponding HEAD

Reply:
Yep that should be easy to handle in router.